src/firefly/js/visualize/ui/ExpandedOptionsPopup.jsx

```diff
@@ -133,59 +138,76 @@ function makeModel(tbl_id,plotViewAry, expandedIds, oldModel) {
 }
 
 function dialogComplete(tbl_id) {
-    const model= getTblById(tbl_id);
+    const model = getTblById(tbl_id);
     if (!model) return;
-    const plotIdAry= model.tableData.data.map( (d) => d[PID_IDX] );
+    const plotIdAry = model.tableData.data.map((d) => d[PID_IDX]);
     if (isEmpty(plotIdAry)) return;
 
-    const currentPlotIdAry= getViewerItemIds(getMultiViewRoot(),EXPANDED_MODE_RESERVED);
-    if (plotIdAry.join('')===currentPlotIdAry.join('')) return;
+    const currentPlotIdAry = getViewerItemIds(getMultiViewRoot(), EXPANDED_MODE_RESERVED);
+    if (plotIdAry.join('') === currentPlotIdAry.join('')) return;
     if (!plotIdAry.includes(visRoot().activePlotId)) {
         dispatchChangeActivePlotView(plotIdAry[0]);
     }
     dispatchReplaceViewerItems(EXPANDED_MODE_RESERVED, plotIdAry, IMAGE);
 }
 
-// const deleteFailedNEW= () => {
-//     getPlotViewAry(visRoot()).forEach( (pv) => {
-//         if (pv.serverCall==='fail') {
-//             dispatchDeletePlotView({plotId:pv.plotId}) ;
-//         }
-//     });
-// };
+const deleteFailedNEW = () => {
+    getPlotViewAry(visRoot()).forEach((pv) => {
+        if (pv.serverCall === 'fail') {
+            dispatchDeletePlotView({plotId: pv.plotId});
+        }
+    });
+};
+
+
+const RemoveSelected = () => {
+    const selectedPlotIds = getSelectedPlotIds();
+    selectedPlotIds.forEach((plotId) => {
+        dispatchDeletePlotView({plotId});
+    });
+}
+
+
+function getSelectedPlotIds(){
+    const {tableModel, selectInfo} = getTblInfoById(TABLE_ID);
+    return selectInfo.selectAll ? tableModel.tableData.data.map((row) => row[2]) : Array.from(selectInfo.exceptions).map((idx)=>
+        tableModel.tableData.data[idx][2]);
+}
```

Comment:
Logic is wrong.  I pretty much wrote the code needed to get selected PlotIds in my comment on your last PR.  Did you not see it?

If you do a select all, then unselect a row before remove images.. it removes the wrong images.